### ISSUE TITLE
configure-saml-authentication is failing for new opsman

### ISSUE THREAD

**Issue by goriparthivamsi** [pivotal-cf/om#572]
Upgraded to latest version of OM CLI 7.4.1 and noticed that SAML authentication is failing for new opsmanager VM. 
Error is as follows: 
om --env env/env.yml --skip-ssl-validation configure-saml-authentication --config config/((redacted))/saml.yml
configuring SAML authentication...
2021/12/27 18:28:51 could not configure authentication: request failed: unexpected response from /api/v0/setup:
HTTP/1.1 422 Unprocessable Entity
Transfer-Encoding: chunked
Cache-Control: no-cache, no-store
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Mon, 27 Dec 2021 18:28:51 GMT
Expires: Fri, 01 Jan 1990 00:00:00 GMT
Pragma: no-cache
Referrer-Policy: strict-origin-when-cross-origin
Server: Ops Manager
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Request-Id: 83d134ae-de9f-41bb-864f-d384cf0748fb
X-Runtime: 0.084294
X-Xss-Protection: 1; mode=block

34
{"errors":{"setup":["IDP Metadata is invalid XML"]}}
0

Switched to older version of CLI which is 7.3.0 and everything is working as expected. I think the issue has been introduced in the latest versions.

**Comment by scottgai** [comment]
My one customer also hit the issue after upgrading to latest version of OM CLI 7.4.1 and noticed that SAML authentication is failing for Ops Manager VM.
Error is as follows:
`om --env env/env.yml --skip-ssl-validation configure-saml-authentication --config config/((redacted))/saml.yml
configuring SAML authentication...
2021/12/27 18:28:51 could not configure authentication: request failed: unexpected response from /api/v0/setup:
HTTP/1.1 422 Unprocessable Entity
Transfer-Encoding: chunked
Cache-Control: no-cache, no-store
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Mon, 27 Dec 2021 18:28:51 GMT
Expires: Fri, 01 Jan 1990 00:00:00 GMT
Pragma: no-cache
Referrer-Policy: strict-origin-when-cross-origin
Server: Ops Manager
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Request-Id: 83d134ae-de9f-41bb-864f-d384cf0748fb
X-Runtime: 0.084294
X-Xss-Protection: 1; mode=block

34
{"errors":{"setup":["IDP Metadata is invalid XML"]}}
0
`
They Switched to older version of CLI which is 7.3.0 and everything is working as expected.

**Comment by scottgai** [comment]
Reproduced the symptom in lab. It started to exhibit since 7.3.2. Probably it's related to [this change](https://github.com/pivotal-cf/om/blob/4d4bea7092973b28a7210c7b9b7326ea0d9a91ce/cmd/loadConfigFile.go#L102) committed to 7.3.2.
For working case, Ops Manager got IDP metadata like this (single backslash '\' to escape double quote)
```
{"setup"=>{"identity_provider"=>"saml", "decryption_passphrase"=>"[FILTERED]", "decryption_passphrase_confirmation"=>"[FILTERED]", "eula_accepted"=>"true", "idp_metadata"=>
"<?xml version=\"1.0\" encoding=\"UTF-8\"?>
......
```
For bad scenario, triple backslash to escape double quote
```
"setup"=>{"identity_provider"=>"saml", "decryption_passphrase"=>"[FILTERED]", "decryption_passphrase_confirmation"=>"[FILTERED]", "eula_accepted"=>"true", "idp_metadata"=>
"<?xml version=\\\"1.0\\\" encoding=\\\"UTF-8\\\"?>
......
```

**Comment by jaristiz** [comment]
Hi @scottgai ,

As you said, it seems that the issue was introduced by a new logic on escaping double quotes, let me review the changes and see how can we work on a fix for this issue.

Also this one is related to #573

**Comment by jaristiz** [comment]
Fixed on version `7.4.2`
